src/deployment/deploy.py

```diff
@@ -323,6 +344,9 @@ def setup_rbac(self) -> None:
                     app.object_id, ApplicationUpdateParameters(app_roles=app_roles)
                 )
 
+            if self.multi_tenant_domain and app.sign_in_audience == "AzureADMyOrg":
+                assign_multi_tenant_auth(app.object_id)
```

Comment:
you should do assignment outside of the if clause so an exisiting deployment can be upgrades to multi tenant

Reply:
Ahhhh, yes. I did not do testing back and forth between the settings on a deployment. I have just completed this. It works great now, even including the failure of the graph API.

Reply:
I suppose this can be done once outside the clause but there is some consistency in the logic since the corresponding action sets the URI here inside the 'if' segment:

            params = ApplicationCreateParameters(
                display_name=self.application_name,
                identifier_uris=[url],
                reply_urls=[url + "/.auth/login/aad/callback"],
                optional_claims=OptionalClaims(id_token=[], access_token=[]),
                required_resource_access=[
                    RequiredResourceAccess(
                        resource_access=[
                            ResourceAccess(id=USER_IMPERSONATION, type="Scope")
                        ],
                        resource_app_id="00000002-0000-0000-c000-000000000000",
                    )
                ],
                app_roles=app_roles,
                
So, for this reason I think we should keep it in both segments even though it appears redundant.